pkg/karmadactl/addons/estimator/estimator.go

```diff
@@ -0,0 +1,162 @@
+package estimator
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	addonsinit "github.com/karmada-io/karmada/pkg/karmadactl/addons/init"
+	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/kubernetes"
+	initutils "github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
+	"github.com/karmada-io/karmada/pkg/karmadactl/utils"
+	"github.com/karmada-io/karmada/pkg/util/names"
+)
+
+// AddonEstimator describe the estimator addon command process
+var AddonEstimator = &addonsinit.Addon{
+	Name:    addonsinit.EstimatorResourceName,
+	Status:  status,
+	Enable:  enableEstimator,
+	Disable: disableEstimator,
+}
+
+var status = func(opts *addonsinit.CommandAddonsListOption) (string, error) {
+	if len(opts.Cluster) == 0 {
+		return addonsinit.AddonUnknownStatus, nil
+	}
+
+	esName := names.GenerateEstimatorDeploymentName(opts.Cluster)
+	deployment, err := opts.KubeClientSet.AppsV1().Deployments(opts.Namespace).Get(context.TODO(), esName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return addonsinit.AddonDisabledStatus, nil
+		}
+		return addonsinit.AddonUnknownStatus, err
+	}
+	if deployment.Status.Replicas != deployment.Status.ReadyReplicas ||
+		deployment.Status.Replicas != deployment.Status.AvailableReplicas {
+		return addonsinit.AddonUnhealthyStatus, nil
+	}
+
+	return addonsinit.AddonEnabledStatus, nil
+}
+
+var enableEstimator = func(opts *addonsinit.CommandAddonsEnableOption) error {
+	if len(opts.Cluster) == 0 {
+		klog.Warning("Cluster is not specified in CommandAddonsEnableOption, estimator installation will skip.")
+		return nil
+	}
+
+	pathOptions := clientcmd.NewDefaultPathOptions()
```

Comment:
Should it is `member-kubeconfig`? `clientcmd.NewDefaultPathOptions()` seems always `~/.kube/config`